src/mono/mono/mini/aot-compiler.c

```diff
@@ -14473,11 +14699,59 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options,
 	}
 #endif
 
-	if (acfg->aot_opts.direct_pinvoke && !acfg->aot_opts.static_link) {
-		aot_printerrf (acfg, "The 'direct-pinvoke' AOT option also requires the 'static' AOT option.\n");
+	if ((acfg->aot_opts.direct_pinvoke || acfg->aot_opts.direct_pinvokes || acfg->aot_opts.direct_pinvoke_lists) && !acfg->aot_opts.static_link) {
+		aot_printerrf (acfg, "The 'direct-pinvoke' flag, 'direct-pinvokes', and 'direct-pinvoke-lists' AOT options also require the 'static' AOT option.\n");
+		return 1;
+	}
+	if (acfg->aot_opts.direct_pinvoke && (acfg->aot_opts.direct_pinvokes || acfg->aot_opts.direct_pinvoke_lists)) {
+		aot_printerrf (acfg, "The 'direct-pinvoke' flag trumps specified 'direct-pinvokes' and 'direct-pinvoke-lists' arguments. Unset either the flag or the specific direct pinvoke arguments.\n");
 		return 1;
 	}
 
+	gboolean added_direct_pinvoke = TRUE;
+	if (acfg->aot_opts.direct_pinvokes) {
+		GList *l;
+
+		for (l = acfg->aot_opts.direct_pinvokes; l; l = l->next) {
+			process_specified_direct_pinvokes (acfg, (char*)l->data);
```

Comment:
added_direct_pinvoke = process_specified_direct_pinvokes (acfg, (char*)l->data); ?

Reply:
Thanks for the catch, I had a commit that removed surfacing the error and then when mulling over it more, decided to surface the error instead and readded the boolean manually, forgetting this important part 😅